diff --git a/rtl8xxxu_8188e.c b/rtl8xxxu_8188e.c
index 7cb5b73..8188e01 100644
--- a/rtl8xxxu_8188e.c
+++ b/rtl8xxxu_8188e.c
@@ -1865,6 +1865,7 @@ struct rtl8xxxu_fileops rtl8188eu_fops = {
 	.tx_desc_size = sizeof(struct rtl8xxxu_txdesc32),
 	.has_tx_report = 1,
 	.init_reg_pkt_life_time = 1,
+	.needs_full_init = 1,
 	.gen2_thermal_meter = 1,
 	.max_sec_cam_num = 32,
 	.adda_1t_init = 0x0b1b25a0,
