diff -ruN a/aic8800_fdrv/aicwf_wext_linux.c b/aic8800_fdrv/aicwf_wext_linux.c
--- a/aic8800_fdrv/aicwf_wext_linux.c	2025-11-03 18:40:02.051876137 -0500
+++ b/aic8800_fdrv/aicwf_wext_linux.c	2025-11-03 20:04:53.916173385 -0500
@@ -417,7 +417,7 @@
 	iwe->cmd = SIOCGIWAP;
 	iwe->u.ap_addr.sa_family = ARPHRD_ETHER;
 
-	if(scan_re->bss && &scan_re->bss->bssid[0]){
+	if (scan_re->bss) {
 	memcpy(iwe->u.ap_addr.sa_data, scan_re->bss->bssid, ETH_ALEN);
 	}
 
diff -ruN a/aic8800_fdrv/rwnx_defs.h b/aic8800_fdrv/rwnx_defs.h
--- a/aic8800_fdrv/rwnx_defs.h	2025-11-03 18:39:07.334501072 -0500
+++ b/aic8800_fdrv/rwnx_defs.h	2025-11-03 19:21:57.309023784 -0500
@@ -70,7 +70,7 @@
 #define HIGH_KERNEL_VERSION KERNEL_VERSION(6, 0, 0)
 #define HIGH_KERNEL_VERSION2 KERNEL_VERSION(6, 1, 0)
 #define HIGH_KERNEL_VERSION3 KERNEL_VERSION(6, 3, 0)
-#define HIGH_KERNEL_VERSION4 KERNEL_VERSION(6, 3, 0)
+#define HIGH_KERNEL_VERSION4 KERNEL_VERSION(6, 4, 0)
 #endif
 
 #ifndef IEEE80211_MAX_AMPDU_BUF
@@ -87,11 +87,19 @@
 #endif
 
 #if LINUX_VERSION_CODE >= HIGH_KERNEL_VERSION
+#ifndef IEEE80211_MAX_AMPDU_BUF
 #define IEEE80211_MAX_AMPDU_BUF                             IEEE80211_MAX_AMPDU_BUF_HE
+#endif
+#ifndef IEEE80211_HE_PHY_CAP6_TRIG_MU_BEAMFORMER_FB
 #define IEEE80211_HE_PHY_CAP6_TRIG_MU_BEAMFORMER_FB         IEEE80211_HE_PHY_CAP6_TRIG_MU_BEAMFORMING_PARTIAL_BW_FB
+#endif
+#ifndef IEEE80211_HE_PHY_CAP6_TRIG_SU_BEAMFORMER_FB
 #define IEEE80211_HE_PHY_CAP6_TRIG_SU_BEAMFORMER_FB         IEEE80211_HE_PHY_CAP6_TRIG_SU_BEAMFORMING_FB
+#endif
+#ifndef IEEE80211_HE_PHY_CAP3_RX_HE_MU_PPDU_FROM_NON_AP_STA
 #define IEEE80211_HE_PHY_CAP3_RX_HE_MU_PPDU_FROM_NON_AP_STA IEEE80211_HE_PHY_CAP3_RX_PARTIAL_BW_SU_IN_20MHZ_MU
 #endif
+#endif
 
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(3, 5, 0) || defined(CONFIG_VHT_FOR_OLD_KERNEL)
diff -ruN a/aic8800_fdrv/rwnx_main.c b/aic8800_fdrv/rwnx_main.c
--- a/aic8800_fdrv/rwnx_main.c	2025-11-03 18:39:31.132664198 -0500
+++ b/aic8800_fdrv/rwnx_main.c	2025-11-03 20:11:49.182549866 -0500
@@ -3844,17 +3844,24 @@
 }
 #endif
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0))
+static int rwnx_cfg80211_set_monitor_channel(struct wiphy *wiphy,
+                                             struct cfg80211_chan_def *chandef)
+#else
 static int rwnx_cfg80211_set_monitor_channel(struct wiphy *wiphy,
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 11, 0))
                                              struct net_device *dev,
-#endif
                                              struct cfg80211_chan_def *chandef)
+#endif
 {
     struct rwnx_hw *rwnx_hw = wiphy_priv(wiphy);
     struct rwnx_vif *rwnx_vif;
     struct me_config_monitor_cfm cfm;
     RWNX_DBG(RWNX_FN_ENTRY_STR);
 
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 1, 0))
+    (void)dev;
+#endif
+
     if (rwnx_hw->monitor_vif == RWNX_INVALID_VIF)
         return -EINVAL;
 
@@ -4377,7 +4384,7 @@
     if (rwnx_vif->vif_index == rwnx_hw->monitor_vif)
     {
         //retrieve channel from firmware
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 11, 0))
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 1, 0))
         rwnx_cfg80211_set_monitor_channel(wiphy, rwnx_vif->ndev, NULL);
 #else
         rwnx_cfg80211_set_monitor_channel(wiphy, NULL);
@@ -8568,4 +8575,3 @@
 MODULE_VERSION(RWNX_VERS_MOD);
 MODULE_AUTHOR(RW_DRV_COPYRIGHT " " RW_DRV_AUTHOR);
 MODULE_LICENSE("GPL");
-
diff -ruN a/aic8800_fdrv/rwnx_tx.c b/aic8800_fdrv/rwnx_tx.c
--- a/aic8800_fdrv/rwnx_tx.c	2025-11-03 18:39:43.130746444 -0500
+++ b/aic8800_fdrv/rwnx_tx.c	2025-11-03 20:04:33.943597962 -0500
@@ -1854,7 +1854,7 @@
 
 netdev_tx_t rwnx_start_monitor_if_xmit(struct sk_buff *skb, struct net_device *dev)
 {
-    int rtap_len, ret, idx, tmp_len;
+    int rtap_len, ret, idx;
     struct ieee80211_radiotap_header *rtap_hdr; // net/ieee80211_radiotap.h
     struct ieee80211_radiotap_iterator iterator; // net/cfg80211.h
     u8_l *rtap_buf = (u8_l *)skb->data;
@@ -2060,8 +2060,8 @@
     }
 
     #if 0
+    int tmp_len = 128;
     // dump buffer
-    tmp_len = 128;
     if (skb->len < 128) {
         tmp_len = skb->len;
     }
diff -ruN a/aic8800_fdrv/rwnx_tx.h b/aic8800_fdrv/rwnx_tx.h
--- a/aic8800_fdrv/rwnx_tx.h	2025-11-03 18:39:55.255829557 -0500
+++ b/aic8800_fdrv/rwnx_tx.h	2025-11-03 19:21:57.311431623 -0500
@@ -171,7 +171,7 @@
                          u64 *cookie);
 #endif /* LINUX_VERSION_CODE >= KERNEL_VERSION(3, 14, 0) */
 #ifdef CONFIG_RWNX_MON_XMIT
-int rwnx_start_monitor_if_xmit(struct sk_buff *skb, struct net_device *dev);
+netdev_tx_t rwnx_start_monitor_if_xmit(struct sk_buff *skb, struct net_device *dev);
 #endif
 int rwnx_txdatacfm(void *pthis, void *host_id);
 
