--- a/BasiliskII/src/Unix/sys_unix.cpp
+++ b/BasiliskII/src/Unix/sys_unix.cpp
@@ -329,7 +329,7 @@
 
			// Parse line
			char *dev = NULL, *mnt_point = NULL, *fstype = NULL;
-			if (sscanf(line, "%as %as %as", &dev, &mnt_point, &fstype) == 3) {
+			if (sscanf(line, "%ms %ms %ms", &dev, &mnt_point, &fstype) == 3) {
				if (strcmp(fstype, "hfs") == 0)
					PrefsAddString("disk", dev);
			}
@@ -509,7 +509,7 @@
			if (strncmp(line, dev_name, strlen(dev_name)) == 0) {
				mount_name[0] = 0;
				char *dummy;
-				sscanf(line, "%as %s", &dummy, mount_name);
+				sscanf(line, "%ms %s", &dummy, mount_name);
				free(dummy);
				fclose(f);
				return true;
