--- a/BasiliskII/src/uae_cpu_2021/cpu_emulation.h
+++ b/BasiliskII/src/uae_cpu_2021/cpu_emulation.h
@@ -88,25 +88,25 @@
 // For HW operations
 // Read/WriteAtariIntXX
 //
-static inline uint64 ReadAtariInt64(memptr addr) {return phys_get_quad(addr);}
-static inline uint32 ReadAtariInt32(memptr addr) {return phys_get_long(addr);}
-static inline uint16 ReadAtariInt16(memptr addr) {return phys_get_word(addr);}
-static inline uint8 ReadAtariInt8(memptr addr) {return phys_get_byte(addr);}
-static inline void WriteAtariInt64(memptr addr, uint64 q) {phys_put_quad(addr, q);}
-static inline void WriteAtariInt32(memptr addr, uint32 l) {phys_put_long(addr, l);}
-static inline void WriteAtariInt16(memptr addr, uint16 w) {phys_put_word(addr, w);}
-static inline void WriteAtariInt8(memptr addr, uint8 b) {phys_put_byte(addr, b);}
+inline uint64 ReadAtariInt64(memptr addr) {return phys_get_quad(addr);}
+inline uint32 ReadAtariInt32(memptr addr) {return phys_get_long(addr);}
+inline uint16 ReadAtariInt16(memptr addr) {return phys_get_word(addr);}
+inline uint8 ReadAtariInt8(memptr addr) {return phys_get_byte(addr);}
+inline void WriteAtariInt64(memptr addr, uint64 q) {phys_put_quad(addr, q);}
+inline void WriteAtariInt32(memptr addr, uint32 l) {phys_put_long(addr, l);}
+inline void WriteAtariInt16(memptr addr, uint16 w) {phys_put_word(addr, w);}
+inline void WriteAtariInt8(memptr addr, uint8 b) {phys_put_byte(addr, b);}
 
 // Direct access to allocated memory
 // Ignores HW checks, so that be carefull
 // Read/WriteHWMemIntXX
 //
-static inline uint32 ReadHWMemInt32(memptr addr) {return do_get_mem_long((uae_u32 *)phys_get_real_address(addr));}
-static inline uint16 ReadHWMemInt16(memptr addr) {return do_get_mem_word((uae_u16 *)phys_get_real_address(addr));}
-static inline uint8 ReadHWMemInt8(memptr addr) {return do_get_mem_byte((uae_u8 *)phys_get_real_address(addr));}
-static inline void WriteHWMemInt32(memptr addr, uint32 l) {do_put_mem_long((uae_u32 *)phys_get_real_address(addr), l);}
-static inline void WriteHWMemInt16(memptr addr, uint16 w) {do_put_mem_word((uae_u16 *)phys_get_real_address(addr), w);}
-static inline void WriteHWMemInt8(memptr addr, uint8 b) {do_put_mem_byte((uae_u8 *)phys_get_real_address(addr), b);}
+inline uint32 ReadHWMemInt32(memptr addr) {return do_get_mem_long((uae_u32 *)phys_get_real_address(addr));}
+inline uint16 ReadHWMemInt16(memptr addr) {return do_get_mem_word((uae_u16 *)phys_get_real_address(addr));}
+inline uint8 ReadHWMemInt8(memptr addr) {return do_get_mem_byte((uae_u8 *)phys_get_real_address(addr));}
+inline void WriteHWMemInt32(memptr addr, uint32 l) {do_put_mem_long((uae_u32 *)phys_get_real_address(addr), l);}
+inline void WriteHWMemInt16(memptr addr, uint16 w) {do_put_mem_word((uae_u16 *)phys_get_real_address(addr), w);}
+inline void WriteHWMemInt8(memptr addr, uint8 b) {do_put_mem_byte((uae_u8 *)phys_get_real_address(addr), b);}
 
 // Indirect access to CPU address space
 // Uses MMU if available
@@ -114,14 +114,14 @@
 // Only data space
 // Read/WriteIntXX
 //
-static inline uint64 ReadInt64(memptr addr) {return get_quad(addr);}
-static inline uint32 ReadInt32(memptr addr) {return get_long(addr);}
-static inline uint16 ReadInt16(memptr addr) {return get_word(addr);}
-static inline uint8 ReadInt8(memptr addr) {return get_byte(addr);}
-static inline void WriteInt64(memptr addr, uint64 q) {put_quad(addr, q);}
-static inline void WriteInt32(memptr addr, uint32 l) {put_long(addr, l);}
-static inline void WriteInt16(memptr addr, uint16 w) {put_word(addr, w);}
-static inline void WriteInt8(memptr addr, uint8 b) {put_byte(addr, b);}
+inline uint64 ReadInt64(memptr addr) {return get_quad(addr);}
+inline uint32 ReadInt32(memptr addr) {return get_long(addr);}
+inline uint16 ReadInt16(memptr addr) {return get_word(addr);}
+inline uint8 ReadInt8(memptr addr) {return get_byte(addr);}
+inline void WriteInt64(memptr addr, uint64 q) {put_quad(addr, q);}
+inline void WriteInt32(memptr addr, uint32 l) {put_long(addr, l);}
+inline void WriteInt16(memptr addr, uint16 w) {put_word(addr, w);}
+inline void WriteInt8(memptr addr, uint8 b) {put_byte(addr, b);}
 
 #ifdef EXTENDED_SIGSEGV
 extern int in_handler;
@@ -159,11 +159,11 @@
 #endif /* EXTENDED_SIGSEGV */
 
 // For address validation
-static inline bool ValidAtariAddr(memptr addr, bool write, uint32 len) { return phys_valid_address(addr, write, len); }
-static inline bool ValidAddr(memptr addr, bool write, uint32 len) { return valid_address(addr, write, len); }
+inline bool ValidAtariAddr(memptr addr, bool write, uint32 len) { return phys_valid_address(addr, write, len); }
+inline bool ValidAddr(memptr addr, bool write, uint32 len) { return valid_address(addr, write, len); }
 
 // Helper functions for usual memory operations
-static inline uint8 *Atari2HostAddr(memptr addr) {return phys_get_real_address(addr);}
+inline uint8 *Atari2HostAddr(memptr addr) {return phys_get_real_address(addr);}
 #endif
 // Possible frame buffer layouts
 enum {
@@ -176,19 +176,19 @@
 
 // Mac memory access functions
 #include "memory.h"
-static inline uint32 ReadMacInt32(uint32 addr) {return get_long(addr);}
-static inline uint32 ReadMacInt16(uint32 addr) {return get_word(addr);}
-static inline uint32 ReadMacInt8(uint32 addr) {return get_byte(addr);}
-static inline void WriteMacInt32(uint32 addr, uint32 l) {put_long(addr, l);}
-static inline void WriteMacInt16(uint32 addr, uint32 w) {put_word(addr, w);}
-static inline void WriteMacInt8(uint32 addr, uint32 b) {put_byte(addr, b);}
-static inline uint8 *Mac2HostAddr(uint32 addr) {return get_real_address(addr);}
-static inline uint32 Host2MacAddr(uint8 *addr) {return get_virtual_address(addr);}
-
-static inline void *Mac_memset(uint32 addr, int c, size_t n) {return memset(Mac2HostAddr(addr), c, n);}
-static inline void *Mac2Host_memcpy(void *dest, uint32 src, size_t n) {return memcpy(dest, Mac2HostAddr(src), n);}
-static inline void *Host2Mac_memcpy(uint32 dest, const void *src, size_t n) {return memcpy(Mac2HostAddr(dest), src, n);}
-static inline void *Mac2Mac_memcpy(uint32 dest, uint32 src, size_t n) {return memcpy(Mac2HostAddr(dest), Mac2HostAddr(src), n);}
+inline uint32 ReadMacInt32(uint32 addr) {return get_long(addr);}
+inline uint32 ReadMacInt16(uint32 addr) {return get_word(addr);}
+inline uint32 ReadMacInt8(uint32 addr) {return get_byte(addr);}
+inline void WriteMacInt32(uint32 addr, uint32 l) {put_long(addr, l);}
+inline void WriteMacInt16(uint32 addr, uint32 w) {put_word(addr, w);}
+inline void WriteMacInt8(uint32 addr, uint32 b) {put_byte(addr, b);}
+inline uint8 *Mac2HostAddr(uint32 addr) {return get_real_address(addr);}
+inline uint32 Host2MacAddr(uint8 *addr) {return get_virtual_address(addr);}
+
+inline void *Mac_memset(uint32 addr, int c, size_t n) {return memset(Mac2HostAddr(addr), c, n);}
+inline void *Mac2Host_memcpy(void *dest, uint32 src, size_t n) {return memcpy(dest, Mac2HostAddr(src), n);}
+inline void *Host2Mac_memcpy(uint32 dest, const void *src, size_t n) {return memcpy(Mac2HostAddr(dest), src, n);}
+inline void *Mac2Mac_memcpy(uint32 dest, uint32 src, size_t n) {return memcpy(Mac2HostAddr(dest), Mac2HostAddr(src), n);}
 
 
 // From newcpu.cpp
@@ -250,11 +250,11 @@
 
 #if 0
 // This function will be removed
-static inline uaecptr showPC(void) { return m68k_getpc(); }	// for debugging only
+inline uaecptr showPC(void) { return m68k_getpc(); }	// for debugging only
 #endif
 
 extern int intlev(void);
-static inline void AtariReset(void) {}
+inline void AtariReset(void) {}
 
 #endif
 
