--- a/meson.build
+++ b/meson.build
@@ -32,7 +32,5 @@ xapian_dep = dependency('xapian-core', static:static_deps)
-if compiler.has_header('mustache.hpp')
+if compiler.has_header('mustache.hpp', args: '-I' + meson.project_source_root() / 'include')
   extra_include = []
-elif compiler.has_header('mustache.hpp', args: '-I/usr/include/kainjow')
-  extra_include = ['/usr/include/kainjow']
 else
-  error('Cannot found header mustache.hpp')
+  error('Cannot find header mustache.hpp')
 endif
